assets/shared/blocks/course-progress/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { sprintf, _n } from '@wordpress/i18n';
+
+/**
+ * Course progress component.
+ *
+ * @param {Object}  props
+ * @param {number}  props.lessonsCount            Number of lessons.
+ * @param {number}  props.completedCount          Number of completed lessons.
+ * @param {boolean} props.hidePercentage          Hide completed percentage.
+ * @param {Object}  props.wrapperAttributes       Wrapper HTML attributes.
+ * @param {Object}  props.barWrapperAttributes    Bar wrapper HTML attributes.
+ * @param {Object}  props.barAttributes           Bar HTML attributes.
+ * @param {string}  props.countersClassName       Counters class name.
+ * @param {string}  props.lessonsCountClassName   Lessons count class name.
+ * @param {string}  props.completedCountClassName Completed count class name.
+ */
+const CourseProgress = ( {
```

Comment:
If supposedly we decide to not include the old classes in the progress bar, I imagine that the progress bars that already exist in site will fail validation correct? So we would need to probably implement a migration using the [deprecation API](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/).

I am asking because as we try to maintain the old classes, we end up with a more complicated component which I am not sure if overall it makes the result cleaner.

Reply:
> If supposedly we decide to not include the old classes in the progress bar, I imagine that the progress bars that already exist in sites will fail validation correct?

It wouldn't be the issue because it's a dynamic block. I kept the old classes to not break sites if the users are adding style customizations based on these classes.